### PR TITLE
Set aside implementation

### PR DESCRIPTION
**Pull request by NBKelly**
I'm making another pull request for this because github appears to be doing something funny with the other one.

I thought the set-aside box wasn't working for one side, but it turned out to actually be one of my browsers caching content so it didn't appear. 

Here's the state of set-aside right now:
* Everything is face-up (the current rules say everything in set-aside is face-up unless otherwise specified - we don't need to worry about it yet)
* It works, and things are removed/returned to the deck (it fixes the bacterial interaction)

Gachapon no longer displays a 'waiting' prompt for the corp. I'm not sure why.

In the future, I think it would be worth implementing tags for visibility, so we can set-aside cards and specify who is allowed to see them face-up.

### CODE REVIEW

**Comment by NBKelly** [comment]
Closes #6324, Closes #6361

**Comment by NBKelly** [comment]
I managed to figure out a solution/technique for setting visibility on set-aside cards.

My solution is:
* When cards are set-aside, we set `corp-can-see` and `runner-can-see` keys in a special `set-aside-visibility` map for the card.
* In `card.cljc`, a card is `set-aside-visible` if it is set-aside, and can be seen by the input side
* By default, cards will be visible to both sides (this is the behavior defined in the rules doc)

As a proof of concept, here's deep dive with the four visibility options:

set-aside-vis-03
set-aside-vis-02
set-aside-vis-01
set-aside-double-vis

The exact implementation could probably be tidier, but I think the concept itself seems reasonable.

File: src/cljc/game/core/card.cljc [hunk 1]
```diff
@@ -144,6 +144,20 @@
...
+  "Checks if the specific card is in a set-aside area."
+  [card]
+  (= (get-zone card) [:set-aside]))
+
+(defn set-aside-visible?
+  "Checks if the specific card is in set aside and visible to this side"
+  [card side]
+  (and (in-set-aside? card)
+       (if (= :corp side)
+         (true? (:corp-can-see (:set-aside-visibility card)))
```
Comment (NoahTheDuke) [inline src/cljc/game/core/card.cljc:157]: Don't need true?

File: test/clj/game/cards/hardware_test.clj [hunk 2]
```diff
@@ -1845,7 +1846,9 @@
       (is (= 3 (count (:deck (get-runner)))) "3 cards remain in deck")
       (is (= 2 (count (:rfg (get-runner)))) "Removed 2 cards from game")
       (is (no-prompt? state :runner) "No more prompts")
-      (is (no-prompt? state :corp) "Waiting prompt cleared")))
+      (is (no-prompt? state :corp) "Waiting prompt cleared")
+      (is false "boop")
```
Comment (NoahTheDuke) [inline test/clj/game/cards/hardware_test.clj:1850]: boop

**Comment by NBKelly** [comment]
I eventually figured out why the gachapon test was failing - the set-aside function was calling effect-completed on the eid given. I believe it shouldn't, but if the correct approach is that we should be making a new eid each time we call set-aside, I can adapt it back to that too.

**Comment by NBKelly** [comment]
I think that covers it for now - I added a couple of extra functions

* `set-aside-for-me` - puts the cards in set aside visible only to their owners (this will save time later)
* `add-to-set-aside` - adds a card into a group of cards already set-aside (given you know the eid)
* `swap-set-aside-cards` - handles swaps where one of the cards (or both) were in set-aside

Additionally, each card in set-aside keeps track of the eid that put it there - this is so you can refer to a card to figure out which group of set-aside cards it belongs to. This is enough so that (later) we could work on changing draw behaviors to use set-aside as in the CR.

**Pull request merged**